api-server.py

```diff
@@ -140,8 +185,57 @@ def main():
         endpoint_rest = "tcp:port={0}:interface={1}".format(args.port_rest, host)
         endpoints.serverFromString(reactor, endpoint_rest).listen(Site(api_server_rest.app.resource()))
 
-    app = Klein()
-    app.run(host, 9999)
+    app = ApiKlein()
+    app.run(host, 9999, syslog_facility=syslog_facility)
+
+    # After the reactor is stopped, gracefully shutdown the database.
+    logger.info("Shutting down - closing databases...")
+    NotificationDB.close()
+    Blockchain.Default().Dispose()
+    NodeLeader.Instance().Shutdown()
+
+
+class ApiKlein(Klein):
+    """
+    ApiKlein extends Klein so that the logging behavior can be customized. Aside from logging,
+    the implementation is identical to Klein.run(): https://github.com/twisted/klein/blob/master/src/klein/_app.py#L376
+    """
+    def run(self, host=None, port=None, endpoint_description=None, syslog_facility=None):
+        if syslog_facility is not None:
+            facility = translate_syslog_facility(syslog_facility)
+            startLogging(prefix="pyapi", facility=facility)
+
+        if not endpoint_description:
+            endpoint_description = "tcp:port={0}:interface={1}".format(port, host)
+
+        endpoint = endpoints.serverFromString(reactor, endpoint_description)
+        endpoint.listen(Site(self.resource()))
+        reactor.run()
+
+
+def translate_syslog_facility(syslog_facility):
```

Comment:
@brianlenz this looks like something that could be very useful in `logzero` itself. what do you think?

Reply:
Yeah, it could be if we want to extend `logzero` to support logging to syslog instead. That makes a lot of sense, so it could be that we refactor this to be more embedded into `logzero` (where it's also reusable) as opposed to here in the API implementation (which I admittedly wasn't a huge fan of in the first place).

Is that where you were going?

Reply:
I think it would make sense, it's useful for many applications.

Reply:
I think the necessity of this refactor is contingent on the outcome of the other discussions above, so let's revisit this once we've got the rest of the behavior squared away?

Reply:
👍